File: api/tmp-build/internal/domain/model/task.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type TaskStatus string

const (
	TaskStatusTodo       TaskStatus = "todo"
	TaskStatusInProgress TaskStatus = "in_progress"
	TaskStatusReview     TaskStatus = "review"
	TaskStatusDone       TaskStatus = "done"
)

type Task struct {
	ID          uuid.UUID       `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	Title       string          `gorm:"type:varchar(255);not null" json:"title" validate:"required,min=3,max=255"`
	Description string          `gorm:"type:text" json:"description"`
	Status      TaskStatus      `gorm:"type:varchar(50);not null;default:'todo'" json:"status" validate:"oneof=todo in_progress review done"`
	Priority    ProjectPriority `gorm:"type:varchar(50);not null;default:'medium'" json:"priority" validate:"oneof=low medium high critical"`
	DueDate     *time.Time      `gorm:"type:date" json:"due_date"`
	ProjectID   uuid.UUID       `gorm:"type:uuid;not null;index" json:"project_id"`
	Project     Project         `gorm:"foreignKey:ProjectID" json:"-"`
	AssignedTo  *uuid.UUID      `gorm:"type:uuid;index" json:"assigned_to"`
	Assignee    *User           `gorm:"foreignKey:AssignedTo" json:"assignee"`
	CreatedAt   time.Time       `json:"created_at"`
	UpdatedAt   time.Time       `json:"updated_at"`
	DeletedAt   gorm.DeletedAt  `gorm:"index" json:"-"`
}

func (t *Task) BeforeCreate(tx *gorm.DB) error {
	t.ID = uuid.New()
	return nil
}

File: frontend/src/types/task.ts
import { Priority } from './project';
import { User } from './auth';

export type TaskStatus = 'todo' | 'in_progress' | 'review' | 'done';

export interface Task {
  id: string;
  title: string;
  description: string;
  status: TaskStatus;
  priority: Priority;
  due_date: string | null;
  project_id: string;
  assigned_to: User | null;
  created_at: string;
  updated_at: string;
}

export interface CreateTaskRequest {
  title: string;
  description?: string;
  priority: Priority;
  due_date?: string;
  assigned_to?: string;
}

export interface UpdateTaskRequest {
  title?: string;
  description?: string;
  status?: TaskStatus;
  priority?: Priority;
  due_date?: string;
  assigned_to?: string;
}
